### ISSUE TITLE
PIMD can't find the unicast route

### ISSUE THREAD

**Issue by miiguelpereiraa** [troglobit/pimd#159]
Hi,

I'm working with the topology below.

![image](https://user-images.githubusercontent.com/48485641/87531278-5f4e2f00-c689-11ea-82ab-62b7f07bf75d.png)

The pimd daemon is running on R1. The S1 is a OVS switch controlled by ONOS, running a PIM app.
In the test that I'm trying to do, both S1 and R1 detect each other has PIM Neighbours, but when R1 (running pimd) receives a PIM Join message at interface 2, sent by S1, it searches the unicast routing table for a route to H2 (10.0.2.1) but it can't find the static route configured (see prints bellow)

![Screenshot from 2020-07-15 10-46-35](https://user-images.githubusercontent.com/48485641/87531710-05019e00-c68a-11ea-85a5-f01b75699f9b.png)

![Screenshot from 2020-07-15 10-48-08](https://user-images.githubusercontent.com/48485641/87531718-0763f800-c68a-11ea-8b62-3d9f001ecb0f.png)

I've tried to configure a direct route to 10.0.2.1 but it didn't work.
I've already tried to change the rp_filter option to "0" at R1 but the problem persisted.
Could you please help me solve this problem?

Thank you very much in advance.

**Comment by troglobit** [comment]
Hmm, that's really odd. Is there something else I should know before trying to replicate?  E.g., host system OS/architecture and pimd version.

**Comment by miiguelpereiraa** [comment]
Hi,

Thank you very much for your response. In the test that I talked about earlier, I was starting pimd with a configuration file that I made where I disabled interface 1 of R1 (because I only wanted to listen to pim messages at interface 2). Has I was trying to debug the problem, I listed the IGMP interfaces in pimd and I noticed that just interface 2 of R1 was listed.
Then I tested running pimd with its default configuration (without using any config file) so that pimd was activated at all interfaces and the problem was resolved. I was able to receive traffic at H1 from the group and source that I was interested in.

I guess I can't disable pimd in interfaces where I will receive multicast traffic.

Best Regards.
